test/CMakeLists.txt

```diff
@@ -30,118 +30,121 @@
-      ${CMAKE_CURRENT_SOURCE_DIR}/OpcodeRequiresCapabilities.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/OpcodeSplit.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/OpcodeTableGet.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/OperandCapabilities.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/Operand.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/OperandPattern.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/SoftwareVersion.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextAdvance.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextDestroy.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextLiteral.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextStartsNewInst.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.Annotation.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.Barrier.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.Constant.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.ControlFlow.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.Debug.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.DeviceSideEnqueue.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.Extension.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.Function.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.Group.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.Image.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.Literal.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.Memory.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.Miscellaneous.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.ModeSetting.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.PipeStorage.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.TypeDeclaration.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextToBinary.SubgroupDispatch.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/TextWordGet.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/UnitSPIRV.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/ValidateFixtures.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/Validate.Capability.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/Validate.CFG.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/Validate.Layout.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/Validate.Storage.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/Validate.SSA.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/ValidateID.cpp
-      ${CMAKE_CURRENT_SOURCE_DIR}/ValidationState.cpp
-    )
-    add_spvtools_unittest(
-      TARGET UnitSPIRV
-      SRCS ${TEST_SOURCES}
-      LIBS ${SPIRV_TOOLS})
-
-    add_spvtools_unittest(
-      TARGET cpp_interface
-      SRCS cpp_interface.cpp
-      LIBS SPIRV-Tools-opt ${SPIRV_TOOLS})
-
-    add_subdirectory(opt)
   else()
     message(STATUS "Did not find googletest, tests will not be built."
       "To enable tests place googletest in '<spirv-dir>/external/googletest'.")
   endif()
 endif()
+
+function(add_spvtools_unittest)
+  if (NOT "${SPIRV_SKIP_TESTS}" AND TARGET gmock_main)
+    set(one_value_args TARGET)
+    set(multi_value_args SRCS LIBS)
+    cmake_parse_arguments(
+      ARG "" "${one_value_args}" "${multi_value_args}" ${ARGN})
```

Comment:
Hmm, I feel that `{one|multi}_value_args` is more readable for me. Otherwise, it seems we need to reference the doc for `cmake_parse_arguments` to know what does each parameter mean.